src/python/dxpy/cli/parsers.py

```diff
@@ -144,7 +144,7 @@ def process_dataobject_args(args):
     process_properties_args(args)
 
     # Visibility
-    args.hidden = (args.hidden == 'hidden')
+    args.hidden = (args.hidden == 'hidden' or args.hidden is True)
```

Comment:
Why are we treating this arg as of a boolean type here?

Reply:
The same args object is used for all files in a single `dx upload` call. But the same args.hidden is also used in `dxpy.upload_local_file(...)` which takes a bool. The conversion is handled here. Before, if you ran this code on args.hidden='hidden' twice, it would return False which is incorrect.

Reply:
OK, got it, thanks for the clarification!